python/feathub/processors/flink/table_builder/join_utils.py

```diff
@@ -142,6 +147,54 @@ def join_table_on_key(
     )
 
 
+def join_lookup_table_on_key(
+    t_env: StreamTableEnvironment,
+    left: NativeFlinkTable,
+    right: NativeFlinkTable,
+    key_fields: List[str],
+) -> NativeFlinkTable:
+    """
+    Lookup join the left and right table on the given key fields.
+    """
+    left = _append_processing_time_attribute(t_env, left)
```

Comment:
Flink SQL API requires that the left table must have a processing time attribute and that the SQL statement must follow the processing time temporal join syntax[1]. Thus I added a processing-time column here.

Despite that we still need to add the column, in this PR's latest update, this method is optimized so the event time attribute is preserved after the column is appended, and there is no DataStream-Table conversion overhead. In this way this method should not have a side effect on Feathub's other functions for now. I have added a TODO so that we may investigate why Flink SQL have such limitations in future.

[1] https://nightlies.apache.org/flink/flink-docs-release-1.16/docs/dev/table/sql/queries/joins/#lookup-join

Reply:
Sounds great.